panel.py

```diff
@@ -308,6 +308,38 @@ def __init__(self, name: str, fdt: Fdt2, node: int) -> None:
 		self.lp11_init = fdt.getprop_or_none(node, 'qcom,mdss-dsi-lp11-init') is not None
 		self.init_delay = fdt.getprop_uint32(node, 'qcom,mdss-dsi-init-delay-us')
 
+		# Display Stream Compression
+		self.compression_mode = fdt.getprop_or_none(mode_node, 'qcom,compression-mode')
+		self.has_dsc = self.compression_mode is not None and self.compression_mode.as_str() == "dsc"
+
+		if self.has_dsc:
+			#self.dsc_lm_split = fdt.getprop_or_none(mode_node, 'qcom,lm-split')
+			dsc_encoders = fdt.getprop_or_none(mode_node, 'qcom,mdss-dsc-encoders')
+			self.dsc_encoders = dsc_encoders.as_int32() if dsc_encoders else None
```

Comment:
Shouldn't these be parsed `as_uint32()`? In https://git.codelinaro.org/clo/la/kernel/msm-extra_group/display-drivers/-/blob/android-13.0.0_r0.122/msm/dsi/dsi_panel.c they're all `read_u32()`.

This means you can just use `fdt.getprop_uint32(mode_node, 'qcom,mdss-dsc-encoders', default=None)`

If you don't need to distinguish between 0 and None you can also omit the default (it's 0 by default).

Reply:
> If you don't need to distinguish between 0 and None you can also omit the default (it's 0 by default).

I guess for most of these the properties are mandatory if there's `dsc` compression mode. In Rust terms, is there a `.unwrap()` on `None` here?

Reply:
If you want to produce an error when these are missing you can just use `.getprop(mode_node, 'qcom,mdss-dsc-encoders').as_uint32()`. The getprop will throw a FDT_ERR_NOTFOUND if missing.

Reply:
By the way it looks like this `dsc_encoders` param is currently completely unused. And useless per what I explained in https://github.com/msm8916-mainline/linux-mdss-dsi-panel-driver-generator/pull/25#discussion_r1349777350. And the number of compression encoders is listed _once again_ in `qcom,display-topology` :upside_down_face:

Reply:
Recap:

- Dropped `dsc_encoders` / `qcom,mdss-dsc-encoders` until we find a use for it and/or `qcom,lm-split` / `qcom,display-topology` upstream (but I hope not);
- Every integer parameter should be mandatory (I cannot think of sensible default values for them... we can always adjust if DT without them shows up);
- The boolean `block-prediction-enable` is obviously not mandatory...;
- DSC (SCR) versions are optional and not always provided.